datalad_crawler/pipelines/loris.py

```diff
@@ -0,0 +1,90 @@
+# emacs: -*- mode: python; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the datalad package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""A pipeline for crawling a LORIS database"""
+
+# Import necessary nodes
+from ..nodes.crawl_url import crawl_url
+from datalad.utils import updated
+from ..nodes.annex import Annexificator
+from datalad_crawler.consts import DATALAD_SPECIAL_REMOTE
+from os.path import basename
+
+
+import json
+# Possibly instantiate a logger if you would like to log
+# during pipeline creation
+from logging import getLogger
+lgr = getLogger("datalad.crawler.pipelines.kaggle")
+
+
+class LorisAPIExtractor(object):
+
+    def __init__(self, apibase=None, annex=None):
+        self.apibase = apibase
+        self.meta    = {}
+        self.repo    = annex.repo
+
+    def __call__(self, data):
+        jsdata = json.loads(data["response"])
+        for candidate in jsdata["Images"]:
+            filename = basename(candidate["Link"])
+            self.meta[filename] = candidate
+            yield updated(data, { "url" : self.apibase + candidate["Link"] })
+        return
+
+    def finalize(self):
+        def _finalize(data):
+            for filename in self.meta:
+                metadata_setter = self.repo.set_metadata(
+                    filename,
+                    reset=self.meta[filename],
+                )
+                for meta in metadata_setter:
+                    lgr.info("Appending metadata to %s", filename)
+            yield data
+        return _finalize
+
+
+def pipeline(url=None, apibase=None):
```

Comment:
So this one is to populate an existing single data set from loris. But wouldn't loris contain many datasets and shouldn't we create many of them to be populated using this pipeline?

Reply:
I should add a comment somewhere. The URL is expected to be the `$LORISAPI/project/$projectname/images` endpoint of the LORIS API, so this pipeline is only extracting a single project.

Reply:
Cool.
Wouldn't it make sense to add another pipeline to create all those? See crcns or openfmri ones

Reply:
Sure, I can add a superpipeline function based on the crcns or openfmri one.